File: src/main/java/fi/metatavu/oioi/cm/rest/translate/CustomerTranslator.java
package fi.metatavu.oioi.cm.rest.translate;

/**
 * Translator for Customer REST entity 
 * 
 * @author Antti Leppä
 */
public class CustomerTranslator extends AbstractTranslator<fi.metatavu.oioi.cm.persistence.model.Customer, fi.metatavu.oioi.cm.model.Customer> {

  @Override
  public fi.metatavu.oioi.cm.model.Customer translate(fi.metatavu.oioi.cm.persistence.model.Customer entity) {
    fi.metatavu.oioi.cm.model.Customer result = new fi.metatavu.oioi.cm.model.Customer();    
    result.setId(entity.getId());
    result.setImageUrl(entity.getImageUrl());
    result.setName(entity.getName());
    result.setCreatorId(entity.getCreatorId());
    result.setLastModifierId(entity.getLastModifierId());
    result.setCreatedAt(entity.getCreatedAt());
    result.setModifiedAt(entity.getModifiedAt());
    return result;
  }
  
}

File: src/main/kotlin/fi/metatavu/oioi/cm/rest/translate/CustomerTranslator.kt
package fi.metatavu.oioi.cm.rest.translate

import fi.metatavu.oioi.cm.persistence.model.Customer
import javax.enterprise.context.ApplicationScoped

/**
 * Translator for Customer REST entity
 *
 * @author Antti Leppä
 */
@ApplicationScoped
class CustomerTranslator : AbstractTranslator<Customer?, fi.metatavu.oioi.cm.model.Customer?>() {

    override fun translate(entity: Customer?): fi.metatavu.oioi.cm.model.Customer? {
        if (entity == null) {
            return null
        }

        return fi.metatavu.oioi.cm.model.Customer(
            id = entity.id,
            imageUrl = entity.imageUrl,
            name = entity.name!!,
            creatorId = entity.creatorId,
            lastModifierId = entity.lastModifierId,
            createdAt = entity.createdAt,
            modifiedAt = entity.modifiedAt
        )
    }
}